tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/series/SeriesReaderByTimestamp.java

```diff
@@ -86,8 +87,10 @@ public Object getValueInTimestamp(long timestamp) throws IOException {
       }
 
       if (data.hasNext()) {
+        Object value = data.currentValue();
         if (data.currentTime() == timestamp) {
-          return data.currentValue();
+          data.next();
```

Comment:
Why not move `Object value = data.currentValue();` to there?

Reply:
Because when getValueInTimestamp() function is called,  its timestamp moves backwards so that point under the same timestamp are taken only once. But data.currentValue() will not move timestamp, so I called data.next() after getting data.currentValue().